docs/wiki/Vision.md

```diff
@@ -0,0 +1,115 @@
+
+In context of bicep or ARM templates we usually leverage a combination of flexible templates that are deployed using different parameter files for different scenarios.
+
+### How do we define a module?
+
+In the context of _CARML_ we define a module as a reusable, template-based building block to deploy Azure resources. As such it is the foundation to apply _Infrastructure as Code_.
+
+By default each module can deploy one instance of a resource and n-amount of its child-resources (for example `1` storage account and `n`-amount of containers). In some instances a module may also deploy strongly coupled resources (for example `1` virtual machine and `n`-amount of data discs).
+
+Each module is generalized for maximum flexibility and optimized for easy usability. The idea is that the template should be able to cover as many resource-specific scenarios as possible and not restrict the user by making assumptions on the users behalf. Eventually the injected parameters should decide what the template does.
+
+Furthermore, each module comes with meaningful default values for it's optional parameters, a detailed documentation for its usage and one or multiple parameter files to proof is correctness.
+
+
+### What is the intended the deployment model?
+
+<img src="media/deploymentModel.png" alt="Deployment model components" height="200">
+
+When working with IaC you use 3 different components:
+- The deployed **environments** that can be individual services, compositions such as workloads, or entire landing zones and the like
+- The **orchestration** deploying the modules in the form of template/pipeline orchestration using for example GitHub actions or Azure DevOps pipelines
+  - No matter the platform we can differentiate two different deployment approaches:
+    - **_Template Orchestration_**: These types of deployments reference individual modules from a 'main/environment' bicep/ARM template and use its capabilities to pass parameters & orchestrate the deployments. By default, deployments are deployed in parallel by the Azure Resource Manager while accounting for all dependencies defined. Furthermore, the deploying pipeline only needs one deployment job that triggers the template's deployment.
+    - **_Pipeline Orchestration_**: This approach uses the platform specific pipeline capabilities (for example pipeline jobs) to trigger the deployment of individual modules, where each job deploys one module. By defining dependencies in between jobs you can make sure your resources are deployed in order. Parallelization is achieved by using a pool of pipeline agents that execute the jobs, while accounting for all dependencies defined.
+- The fundamental **modules** that each deploy a service, i.e. are the building blocks to deploy environments
+
+To make each component a bit more tangible, let's take a look at the following example:
```

Comment:
I believe this section is not intended for the vision document it is too detailed, so maybe can be pushed to the bottom when you talk about the 'how'

Reply:
Moved it to the section with the detailed flow diagram